tests/shtest

```diff
@@ -438,39 +438,86 @@ cmp $d/color $d/expect
 ## Default colors, complex input
 $JQ -Ccn '[{"a":true,"b":false},123,null]' > $d/color
 {
-  printf '\033[1;39m[\033[1;39m{'
-  printf '\033[0m\033[1;34m"a"\033['
-  printf '0m\033[1;39m:\033[0m\033['
-  printf '0;39mtrue\033[0m\033[1'
-  printf ';39m,\033[0m\033[1;34m'
-  printf '"b"\033[0m\033[1;39m:\033'
-  printf '[0m\033[0;39mfalse\033'
-  printf '[0m\033[1;39m\033[1;39'
-  printf 'm}\033[0m\033[1;39m,\033['
-  printf '0;39m123\033[0m\033[1;'
-  printf '39m,\033[0;90mnull\033'
-  printf '[0m\033[1;39m\033[1;39'
-  printf 'm]\033[0m\n'
+  printf '\033[1;39m[\033[0m'
+  printf '\033[1;39m{\033[0m'
+  printf '\033[1;34m"a"\033[0m'
+  printf '\033[1;39m:\033[0m'
+  printf '\033[0;39mtrue\033[0m'
+  printf '\033[1;39m,\033[0m'
+  printf '\033[1;34m"b"\033[0m'
+  printf '\033[1;39m:\033[0m'
+  printf '\033[0;39mfalse\033[0m'
+  printf '\033[1;39m}\033[0m'
+  printf '\033[1;39m,\033[0m'
+  printf '\033[0;39m123\033[0m'
+  printf '\033[1;39m,\033[0m'
+  printf '\033[0;90mnull\033[0m'
+  printf '\033[1;39m]\033[0m\n'
 } > $d/expect
 cmp $d/color $d/expect
 
 ## Set non-default colors, complex input
 JQ_COLORS='0;30:0;31:0;32:0;33:0;34:1;35:1;36:1;37' \
   $JQ -Ccn '[{"a":true,"b":false},123,null]' > $d/color
 {
-  printf '\033[1;35m[\033[1;36m{'
-  printf '\033[0m\033[1;37m"a"\033['
-  printf '0m\033[1;36m:\033[0m\033['
-  printf '0;32mtrue\033[0m\033[1'
-  printf ';36m,\033[0m\033[1;37m'
-  printf '"b"\033[0m\033[1;36m:\033'
-  printf '[0m\033[0;31mfalse\033'
-  printf '[0m\033[1;36m\033[1;36'
-  printf 'm}\033[0m\033[1;35m,\033['
-  printf '0;33m123\033[0m\033[1;'
-  printf '35m,\033[0;30mnull\033'
-  printf '[0m\033[1;35m\033[1;35'
-  printf 'm]\033[0m\n'
+  printf '\033[1;35m[\033[0m'
+  printf '\033[1;36m{\033[0m'
+  printf '\033[1;37m"a"\033[0m'
+  printf '\033[1;36m:\033[0m'
+  printf '\033[0;32mtrue\033[0m'
+  printf '\033[1;36m,\033[0m'
+  printf '\033[1;37m"b"\033[0m'
+  printf '\033[1;36m:\033[0m'
+  printf '\033[0;31mfalse\033[0m'
+  printf '\033[1;36m}\033[0m'
+  printf '\033[1;35m,\033[0m'
+  printf '\033[0;33m123\033[0m'
+  printf '\033[1;35m,\033[0m'
+  printf '\033[0;30mnull\033[0m'
+  printf '\033[1;35m]\033[0m\n'
+} > $d/expect
+cmp $d/color $d/expect
+
+## Default colors, complex input, indented
+$JQ -Cn '[{"a":true,"b":false},123,null]' > $d/color
+{
+  printf '\033[1;39m[\033[0m\n'
+  printf '  \033[1;39m{\033[0m\n'
+  printf '    \033[1;34m"a"\033[0m'
+  printf '\033[1;39m: \033[0m'
+  printf '\033[0;39mtrue\033[0m'
+  printf '\033[1;39m,\033[0m\n'
+  printf '    \033[1;34m"b"\033[0m'
+  printf '\033[1;39m: \033[0m'
+  printf '\033[0;39mfalse\033[0m\n'
+  printf '  \033[1;39m}\033[0m'
+  printf '\033[1;39m,\033[0m\n'
+  printf '  \033[0;39m123\033[0m'
+  printf '\033[1;39m,\033[0m\n'
+  printf '  \033[0;90mnull\033[0m\n'
+  printf '\033[1;39m]\033[0m\n'
+} > $d/expect
+cmp $d/color $d/expect
+
+## Set non-default colors, complex input, indented
```

Comment:
Does this regression test the reset issue for arrays? i can visually see the issue using `JQ_COLORS='0;30:0;31:0;32:0;33:0;34:7;35:1;36:1;37' jq -Cn '[{"a":true,"b":false},123,null]'` (7 to reverse colors for arrays)

Otherwise nice with more tests 👍

Reply:
![file](https://github.com/jqlang/jq/assets/20175435/369d3407-441f-4efa-9487-36ef87a71a72)